### DISCUSSION TITLE
Fahrprofil_0

### DISCUSSION THREAD

**Discussion by Mjukes** [jilleb/MQB-FPA#83] [Q&A]
Hi,  I hope you can help.  I've got a 2020 Passat GTE with DCC and Ambient plus.  I have assigned colours to my drive modes,  but whilst researching this,  I found many references to Fahrprofil_0 being "coming/leaving home"  and having a different colour assigned to it.

However assigning a colour to this profile on my Passat does nothing.

I'm essentially looking to have the Ambient strips come on with CH/LH and this profile appears to control that.  Looking through your data you don't seem to apply too much to Profile 0,  I wondered if you knew its role and if it could be activated on a car missing it?

Many thanks in advance

Mike

**Comment by jilleb** [comment]
Hi,

Good question, let's test if profile 0 does something.

**Reply by Mjukes** [reply]
Excellent. Thank you.

**Reply by Mjukes** [reply]
in case it helps my BCM is 5Q0937086BR running SW 303.

**Comment by Cuzoe** [comment]
I have not done anything with profile zero in the FPA data set. But my ambient lights come on when I unlock, and they are on when I turn off ignition.  I'm assuming that is what you mean.

That is set in BCM adaptations.  I have mine set for blue. Once the ignition turns on the ambient lights to go to the color for whatever driving profile I am in, after ignition is off they go back to blue.

**Reply by Mjukes** [reply]
This is exactly the behaviour i seek. 
Do they also stay on with your coming home lights?

Thank you

**Reply by Cuzoe** [reply]
I will have to check if they remain on with coming home lights.  Normally I am walking away up the stairs as soon as the garage door closes, haha.

Maybe post an adaptation map of your BCM, could have a quick look.  Curious about the newer BCMs that you and @jilleb mention.

**Reply by Mjukes** [reply]
Haha. That’s fair enough. It’s clear that colour change behaviour with unlocking and ignition is different. 


I won’t lie, I only have OBDeleven at this time so here is the output it gives. I hope it helps. 
[BeforeMODOBDeleven-Log.txt](https://github.com/jilleb/MQB-FPA/files/10499704/BeforeMODOBDeleven-Log.txt)

**Reply by Cuzoe** [reply]
Okay, try this...
Int. light, 2nd generation
     Defaultwert Ambienteprofil Farbe
		0

**Change from 0 > 2.**

This is based on my car... Normal profile = 2, as usual.  And I have the color for Normal set to blue.
_That's in the "Ambientelicht Zuordnung der Farbe zum Fahrprofil" channels._  

So my "default" is actually the color from normal.  I think this is what I ended up doing because Fahrprofil 0 didn't seem to really do anything.

**Reply by Mjukes** [reply]
Oh man you had my hopes up there. Unfortunately it doesn’t make any difference. Lights still go out as soon as the car is locked and come on when the door is open with the last selected profile colour 😞

**Reply by jilleb** [reply]
I have another thing i want to test...fingers crossed

**Reply by Mjukes** [reply]
🤞🏻🤞🏻🤞🏻🤞🏻🤞🏻🤞🏻🤞🏻🤞🏻🤞🏻🤞🏻🤞🏻

**Reply by Cuzoe** [reply]
A few of your "conditional" brightness channels for "unlocked" are set to 0%, for example...
helligkeit entriegelt tueren - 0%
Translates, roughly, to "brightness of doors when car unlocked"

Fussraum (footwells) and itafel (instrument panel)are also set to 0%.

Maybe set those to 100% and see what happens.  


I'm not actually sure the ambient light is tied to coming home and leaving home necessarily. Leaving home lights just happen to correspond with unlocking the car.  But coming home lights are still on when the car is locked.  

That being said, I don't suppose there's much need for the ambient lights to stay on if you've locked the car and have walked away, no one is going to be able to see them anyway. Of course your coming home exterior lights will still be on for whatever time you have set.

**Reply by Mjukes** [reply]
Holy shit

you’ve half cracked it. 

Changing helligkeit entriegelt tueren to 95% has the door coloured strips come on as soon as I unlock as opposed to when I open the door. This is huge as it was one of the main asks I had (I.e. walking up to the car and seeing the colour)

it is still the same colour as the last profile selected which isn’t the behaviour I would like in a perfect world but it’s a start. 

I dont suppose you know which of those paramenters would relate to the dash strip?

**Reply by Cuzoe** [reply]
It really depends on which part number LED is being used for the dash, and what group it's assigned to... that gets a little more into the weeds... happy to go there if you care, haha.

But there are 7 groups (8 sort of, if you count panaromic sunroof)
tueren - doors
fussraum - footwells
cockpit - cockpit
mittelkonsole - middle console
dach - roof
lautsprecher - loudspeaker 
i tafel - instrument panel

In your adaptations I see door, footwell and instrument panel were set to 0%.  So maybe the RGB for your dash is in the instrument panel group...

_p helligkeit entriegelt itafel_... Change from 0% to whatever you want.
_p helligkeit entriegelt fussraum_... Change from 0% to whatever you want, if/assuming you have RGB footwells

**Reply by Cuzoe** [reply]
As I'm thinking about it, I'm always driving in "Individual" and my color setup for individual profile is blue LEDs.  There is this adaptation
_Ambiente Farbwahl FPA waehlbare Kopplung Status hmi default
		**coupled**_

Translates to... Ambient color choice FPA selectable coupling status hmi default

When set to coupled the ambient lights will match the driving profile assignments.  If you go to the ambient light page on the infotainment you may have an option at the top right... Off, Automatic, Manual.  (For some reason my car does not show this, can't figure out why not).  But Automatic in this menu means colors with driving profile.  If you have this set to Automatic that might be why you see the driving profile color when you unlock the car.

Try changing from Automatic > Manual (if you have the option).

Otherwise you could set the adaptation to "not coupled" (or whatever the option is then play around.  

Probably a good idea, after making any of these changes, to go into Vehicle Menu and reset ambient colors to default settings. 
 Sometimes the settings get "stuck" even though the BCM accepts changes and shows results immediately.

**Reply by Mjukes** [reply]
Thanks for this. I do have the auto and manual menu but as I understand it, if I select manual it won’t change when I change modes. So a trade off?
So far as coupled goes, I understood that to mean the colours on the stereo/cockpit following the  profile changes. 

I’ll try those two 0% settings to see what they effect. One quick Q, I’ve never seen the ‘reset default colours’ menu option?  Where does that live?

**Reply by Mjukes** [reply]
Ok. Success. Instrument is the dash light. So I now have strips on with unlock WHICH IS HUGE

now what are the odds the the ignition off profile is simple an adaptation…?

**Reply by Cuzoe** [reply]
Vehicle menu on the infotainment, then scroll down to the bottom I think, option to reset default settings... Then inside of that there are several options, ambient lighting should be one of them.

I don't know if I would call the Manual/Automatic a trade-off.  It just gives you choices.  Maybe you really like green and that's the color that you always want for your ambient lighting.  So you would select Manual and then your lighting would stay green no matter the driving mode.  And then if you decide you want ambient lighting to change with driving mode you set it back to Automatic.

That adaptation only determines the Manual/Automatic setting anytime you do a default settings reset of ambient lighting.  Once you have made a selection on your own, the system remembers that setting always.

So in your car right now if you do a reset it's going to be set to Automatic.  But if you then change it to Manual (or Off) it will remember that setting indefinitely.

The "coupling" of ambient lighting and infotainment/dash accent color is a function of the two color lists in your BCM adaptations.  
- Notice one of them just says color list (1-30)... that one determines the color of infotainment/dash.  
- The other one says color list LIN (1-30)... That one determines the color of the LEDs for ambient lighting

When you select, for example the 8th color in your slider, you are selecting Color List 8 and Color List LIN 8.
@jilleb has a post on drive2 that talks about this.  But in summary an LED screen will show the exact color of those RGB settings.   Physical LEDs do not because red green and blue diodes have different brightness when set at the same "level."

So let's say color number 5 is purple.  Color List may have this as Red - 255, Blue - 255, Green - 0.  Color List LIN might have Red - 255, Blue - 215, Green - 0.  I don't know the actual mix, but in that example blue diodes are brighter. So in order to actually get purple ambient lights you have to turn down blue level.

Without the two lists, the colors of your ambient lights wouldn't match the color on the screens.

**Reply by Mjukes** [reply]
Awesome. I’d never touched that reset to default settings as I’d assumed it was system wide. I’ll run that to see if it helps

In the meantime (prior to trying a reset) switching to manual and rebooting the stereo. Then turning car off. Lock. Unlock does not display an alternative colour. 

I’ll have to try the rest tomorrow as wife is getting grumpy me keep running out. Lol

I do think that alternative colour lives exclusively within the Fahrprofil_0 as a seperate process when unlocking the car. 

Keen to understand that more. 

Thank you again

**Reply by Cuzoe** [reply]
Make sure you go into the vehicle menu first, and settings default settings. Not the settings, menu and default settings.

In either case, I think there's a prompt that says are you sure you want to do this, haha.

When I get a chance I'll play around in my car and see what the lights do. But remember for whatever reason I can't get the automatic and manual option.  But I have a Frankenstein car sort of, didn't come with any RGB lighting and the facelift BCM that is capable of RGB lighting has been retrofitted.

One cool thing that I do have set up (that was not my idea, taken from somebody on drive2, can't remember who at the moment)... Is that color list and color list LIN don't actually have to match.  In my car I have sort of a holiday setting 😂.  Color list 15 is set to show green... So that's infotainment and dash.  And that's the color I see in spot number 15 of the slider.  But color list LIN 15 is set to red.

So when I go to position 15 on the slider I get green infotainment and dash but red ambient lighting.  Happy holidays in my ride 😁.

**Reply by Mjukes** [reply]
Hahaha. I love that!!  Mate you’ve been so helpful. Thank you again. 

I’m
Hoping I do that reset and all sort of crazy shit starts happening. Lol

**Reply by Cuzoe** [reply]
So I just tried in my car and indeed, when unlocked the ambient lights are coming on in the color for the current driving mode.

This is completely fine with me. But I suppose that's not how it should work so I'll have to mess around when I get time.

There was one point when I first started coding for RGB LEDs that I had them coming on red when I unlocked the car.  But at that time I had so many things that weren't working the I don't remember what change(s) stopped that specific behavior.

**Reply by Mjukes** [reply]
If the colour is assigned to the profile, the car essentially needs to switch profiles automatically with ignition off and remember the ignition on state. 
Considering we can have them on with ignition off it’s this switch (and potential lack of profil_0 data) that is not taking place. 

Im going to see if I can compare my adaptations to the ones @olli991 shared yesterday to see if there are any obvious differences.

**Reply by Mjukes** [reply]
Had a look through and couldn’t see anything
Obvious that might be different. 

Also, had a look for the reset colours option and there’s nothing there. 

Vehicle
Settings
Reset default settings

which then gives a warning saying “settings will be reset are you sure. “

🧐

**Reply by Cuzoe** [reply]
I won't have a chance to mess with any profiles or adaptations this weekend but I'll get a picture of the reset menu this afternoon.

**Reply by Mjukes** [reply]
Cheers mate 👍🏼

**Reply by Mjukes** [reply]
I double checked the menus and there's no reset option at all in the vehicle menu,  only in the main menu and it's a very definite full reset.
It's a 2020 Passat GTE

I don't think it's too critical to be honest.

Any further progress  @jilleb  ?

**Reply by Cuzoe** [reply]
I suppose I don't know what menus you have... but there is definitely an ambient light reset menu, among other reset menus.  I cannot imagine your car doesn't have at least some of them.

Vehicle Menu > Settings button at the bottom right > Scroll down to "Factory Settings" > Scroll down to "Ambient Lighting" > Reset just the ambient lighting.

![Page 1](https://user-images.githubusercontent.com/101527335/214369515-83f8a745-3193-47f7-b5b8-5b0af705b0a3.JPG)
![Page 2](https://user-images.githubusercontent.com/101527335/214369540-e9fa29d1-8208-48a1-adb7-828feb03219c.JPG)
![Page 3](https://user-images.githubusercontent.com/101527335/214369554-47092e17-c2a2-4d04-baf1-862932c2acde.JPG)
![Page 4](https://user-images.githubusercontent.com/101527335/214369576-7c4047d7-2688-4e7c-9b6b-535d1e737cc6.JPG)

And you may not think it's critical... your decision.  I only speak from experience on my car, the same experience that now has your door and dash lights coming on with door unlock, haha.

**Reply by Mjukes** [reply]
![A857D182-A4C4-41C4-8911-A9B22ABA44DE](https://user-images.githubusercontent.com/122812996/214382370-2dfe44af-9b01-4932-a579-0e3359a1017c.jpeg)
![9A9DA45B-A95C-4316-AD6B-355DB5F0F28D](https://user-images.githubusercontent.com/122812996/214382389-cc951d22-76f4-4e95-8753-a8504e11b59c.jpeg)
Haha. 

No I mean. I want it to be critical. Lol

here’s what I have. 

![Uploading 153AA32D-F21F-4FAE-B1BD-870A457960A8.jpeg…]()

**Reply by Cuzoe** [reply]
Your picture isn't loading for me, although it does look like something is attached...
![image](https://user-images.githubusercontent.com/101527335/214381283-21e3596d-7bab-4656-8cb6-5eafa1f6edfe.png)

Whenever I mess around with colors, especially given that my car does not show the Manual/Automatic option... I have to do an Ambient Light reset to get things back to normal.

**Reply by Mjukes** [reply]
Fixed

**Reply by Cuzoe** [reply]
That top picture is the wrong menu (looks like you're in system settings and not vehicle settings).  

In that lower picture, once you hit settings you aren't able to scroll all the way down?

**Reply by Mjukes** [reply]
There’s no other options even remotely similar on either menu. Must be a model thing. 
I checked and double checked. It is weird, the way VW just remove stuff is mind numbing.

**Reply by Cuzoe** [reply]
What menu do you get when you press the settings button at the bottom right of the vehicle status screen?

Does that go to vehicle settings or system settings?  In my car it goes to vehicle settings.

**Reply by Mjukes** [reply]
That goes to vehicle. Or background lighting etc. I’m going to go and check the personalisation menu…

**Reply by Cuzoe** [reply]
In vehicle menu you can't scroll?

The very last thing in my vehicle menu is factory settings. Once I select that I get a new set of options that includes ambient lighting, lighting, wipers, parking, etc.

**Reply by Mjukes** [reply]
Yeah. I can scroll. Just not there. Checked personalisation menu and there’s nothing there. I’ve asked in the forums if it’s hidden or anything.

**Reply by Cuzoe** [reply]
Strange, I've always had that menu there... Going back to when I picked it up from the dealer.  That was countless retrofits ago so there were fewer things I could factory reset.  Possibly only lights, opening and closing and mirrors/wipers.

But back then one of the things that was always talked about on the forums was that if you made changes, typically to things like DRLs... You should go into this menu and do the reset of Lights.

Where are these options in your car now?  Maybe post an admap of 5F Infotainment. Or better yet attach it. That way it doesn't make the thread so long to scroll through, haha.

**Reply by Mjukes** [reply]
Lol. Let me see if I can pull that in the morning. 👍🏼

**Reply by Mjukes** [reply]
So,  I suspect this is another quirk of the 2020,  which features the E-sim and Driving profiles where your settings are saved to the profile.  Makes no sense and I have looked at all menus and cannot find anything that allows individual resetting.  The 2018 Passat DID have this menu (confirmed by other owners)
I've attached my adaptation map (I hope).  Perhaps it's a menu that can be activated?

[Multimedia.txt](https://github.com/jilleb/MQB-FPA/files/10499630/Multimedia.txt)

**Comment by jilleb** [comment]
I added data to profile 0 of the fpa dataset. The behavior doesnt change. I know older bcm versions support a different color when unlocking... But so far on the newer ones, I havent reproduced it.

**Reply by Mjukes** [reply]
I had read it had been ‘removed’ from the later BCM’s but I couldn’t understand why.  Really hope you find something.

**Reply by olli991** [reply]
It is working on older version. In my car it's working.
I'm running:
SW 0273
HW H38
Partn-No. 5Q0937084DD
HW-No. 5Q0937084CG

Attached is my current adaptaion of the bcm.
[BCM adap.txt](https://github.com/jilleb/MQB-FPA/files/10472482/BCM.adap.txt)

**Reply by Mjukes** [reply]
What car/year?  I can see your fahrprofil_0 has a number assigned so assume this is standard?  Or did you make changes to get it working?

**Reply by olli991** [reply]
It's all changed by me. It's a Seat Leon ST MY2019. 
I have changed it to 30 colors and made the 30th color white and coded that to profile 0

**Reply by Mjukes** [reply]
See I think that’s the difference. My profil-0 does nothing no matter what colour I select.

**Comment by lito69** [comment]
On Seat leon BCM works profile 0 on CG,CQ,DD, more people downgrade DH to DD for this question and others opction not abaliable on DH

**Reply by Cuzoe** [reply]
Ahhhh... that makes sense to why it used to work in my car (NAR Golf 7, facelift BCM retrofit). 

When I first did facelift BCM I used 5Q0937084DD, bought from EU.  I had to change because NAR key is 315Mhz, while EU is 433/434.  There is an adaptation in the BCM to choose frequency.  It seems to change the receiver frequency but the antenna in the EU BCM is not tuned for 315.  My remote key only worked if held right beside the BCM (I do not have KESSY).

Anyway, seems that it does not work now, my BCM is 5Q0937085CC, from some Audi... 
Maybe I'll try proper Mk7.5 NAR BCM 5Q0937087BB next :D

**Comment by Mjukes** [comment]
Definitely seems to be at BCM level. Quite why they would decide to remove that behaviour god only knows.

**Reply by Mjukes** [reply]
Ok,  so heres an interesting quirk/behaviour

So,  on the Passat GTE whatever driving profile you select is remembered when you restart the car.

Except GTE mode.  In GTE mode,  once you switch the engine off it reverts to the mode you'd chosen prior to pressing GTE.

Except it doesn't entirely

With my adaptations, GTE mode seems to adopt whatever colour I select when I choose Manual in the ambient Menu.  I am not sure which Fahrprofil actually relates to GTE mode.  In this case I use RED

However,  if I am in GTE mode when I shut my car off,  when I then unlock (thanks @Cuzoe ) the door strips come on.... RED.

And they stay red until I turn on the ignition...whereby they change to the previously selected mode. (Purple)

The "manual" / GTE connection is odd and not one I have explored.  Is there a way to see which profile GTE lives under using OBD11?

I wonder if this switching behaviour is useful to us at all?

**Reply by Cuzoe** [reply]
GTE will be it's own driving mode (which most cars don't have, obviously).

The behavior you describe with the car not "staying" in a mode is normal, and is part of the FPA dataset.  The dataset actually dictates which Modes the car is **allowed** to start (after ignition cycle) in, and even further dictates which mode some individual modules are **allowed** to start (after ignition cycle) in. 

One example is that on the Golf R the exhaust flaps always start in Normal (mid way between open/close).  It does not matter if the person turned the car off in Race mode, where the exhaust flaps were full open.  On ignition cycle, they will go back to mid range, even though the driving mode is still Race.  Similar for the DSG, which does not start in Sport mode.   The FPA dataset effectively has some defaults.

You can modify your dataset, as is the subject of this GitHub, to change this behavior.  I have modified my dataset to **allow** everything to start in every mode... which means for me nothing changes between ignition cycles.

**Comment by Mjukes** [comment]
So you think GTE will live outside of the 15 Farhprofils?

**Reply by Cuzoe** [reply]
GTE Mode is the weird one, but nothing lives outside of the 15, Probably Profile 8.  In the attached picture it shows as "unknown" but on some other GTE datasets this is Profile 8...

This if from a 2021 Passat GTE, I wonder if this is a factory error in the dataset
![Passat 2021 GTE](https://user-images.githubusercontent.com/101527335/214916880-7931b5fc-dee1-4898-a1c2-1ce6ba6d1bf2.JPG)

This if from a Magotan GTE (Chinese Market Passat equivalent).  And GTE mode is set to Profile 8.
![Magotan (Chinese Market Passat) GTE](https://user-images.githubusercontent.com/101527335/214917680-5af42035-f727-4573-be76-28bd076067a8.JPG)

**Comment by Mjukes** [comment]
Ahh. That sort of makes sense. I have the individual
Profile set in Int. light 2nd generation set as 8, which is blue. 
So perhaps it’s overwriting profile 8 with whatever is set in the manual (individual?) ambient setting?

**Reply by lito69** [reply]
Yes

**Comment by Mjukes** [comment]
Right. That’s resolved that. The behaviour above is exactly what’s happening. I’ve now set the ‘individual’ (manual) to profile 0. And GTE then reports the correct colour. 

I had hoped setting to 0 might stir something but nothing

So again back to dataset as to how it behaves on those earlier BCM’s

the data you shared above. Do you have any examples of what 0 looks like on a some cars?

**Reply by Cuzoe** [reply]
I'm not sure exactly what you're asking... And I'm not on my computer now.  But if I sent a screenshot like above, the top line would say...

_Profile Position 0: Comfort (1)_

**Comment by Mjukes** [comment]
Thanks. Maybe I’m misunderstanding. My Fahrprofil 1 is comfort. 

I’ve always assumed that profile 0 (the colour that is called when ignition off) was a profile in its own right. So wondered if you could see the engine, gearbox etc data for that profile 0?

**Reply by Cuzoe** [reply]
I understand what you're asking, but there is no Profile 0 as you're thinking of it... keep in mind a Profile 0 for most of those things (engine, gearbox, etc) doesn't make sense.  Engine for example, is either in Normal, Eco or Sport... when the ignition is off there would be no engine setting.

There are lot of possible profiles that can be assigned... https://github.com/jilleb/MQB-FPA/wiki/dataset#0x32d---0x339-profiles
You'll see that up to 12 (0-11, or 1-c in HEX) of those can be configured to show,  But Profile 0 is "not set" which is probably why getting a certain color RGB light when unlocking the doors is entirely a BCM function (and was, clearly, broken in later BCM software).

The "Profile 0" data isn't coming from the driving profile data set.  It's really just the BCM software/adaptations referencing itself.

**Reply by Mjukes** [reply]
Thanks mate. That helps me understand better. I had a vision of a blank profile with ignition driven switching

You getting me lights on unlock was huge, I’d love the colour change but it’s not looking likely. Lol

**Reply by jilleb** [reply]
I think we need to file a bug report at VW 😁
Hey VW, we tested your software and a feature is missing.

**Reply by Mjukes** [reply]
Lol. I think what bugs me is someone has removed the behaviour but I can’t understand why. If there was some logic behind, it would be Easier to accept.

**Reply by Cuzoe** [reply]
I think it's pretty unlikely that someone "removed" the behavior. It was probably just broken while making other changes. And then obviously wasn't caught.

One mistake in the code anywhere can break something.  That's not to excuse it, but on balance it's a trivial feature.

And maybe they did realize there was a bug but too late to go back and modify the software. And that's why in your adaptation those channels were set to 0% brightness on unlock, so you would never know about the bug anyway 😂.

**Reply by Mjukes** [reply]
It could perhaps be tied to the E-sim
And user profile system the MIB3 unit users and like you say it may not have been compatible and so they just rubbed it out. 

I assume then it’s not something we are likely to find and resolve. 😞

**Reply by Cuzoe** [reply]
I had user profiles on MIB1, 2 and now 2.5 also.  Problem here is the BCM, not the MIB.  The user profiles are able to save a driving mode. For instance, my wife gets in the car and when she selects her user profile the car switches to comfort mode, if that's the last driving mode she was in.

The ambient light colors are then a function of the driving mode, not her profile.  It's possible that differs if you are in Manual color mode.  But I wouldn't bet my money on that.  Just a simple bug that the color selected for "Profile 0" is not actually being called by the BCM on unlocked.

**Reply by Mjukes** [reply]
Ah. That makes sense. Well it’s been fun squashing bugs 😡 😂

**Reply by lito69** [reply]
When turning on the color changes to the mode that loads if this ambient light is automatic, in manual ambient light it will surely load the color of the customization because it will not send color associated with the profile.

Profile 0 I think is the state where it does not receive data from the profile gw, when there is no ambient light auto mode there is no profile and it is placed in the color assigned in manual mode.

You could put here the dataset that has worked, someone with gte wanted to go from gw 5q0 to 3q0 and I needed it

**Reply by Mjukes** [reply]
Is your pre-ignition colour dictated by this line

LO]_Defaultwert Ambienteprofil Farbe:	2 ?
